### PR TITLE
Add requirements.txt where missing and delete unused deps

### PR DESCRIPTION
**Pull request by sevein**
#### dashboard:
- debian/control
  - \- python-dateutil **(moved to requirements.txt)**
  - \- python-gearman **(moved to requirements.txt)**
  - \- python-simplejson **(unused)**
- requirements.txt
  - \+ gearman==2.0.2
  - \+ python-dateutil==2.4.2
#### MCPClient:
- debian/control
  - \- python-lxml **(moved to requirements.txt)**
  - \- python-gearman **(moved to requirements.txt)**
  - \- python-mysqldb **(moved to requirements.txt)**
  - \- python-unidecode **(moved to requirements.txt)**
- requriements.txt
  - \+ bagit==1.5.2
  - \+ Django>=1.7,<1.8
  - \+ django-annoying==0.7.7
  - \+ django-autoslug==1.7.1
  - \+ django-mysqlpool==0.1-9
  - \+ django-extensions==1.1.1
  - \+ MySQL-python==1.2.5
  - \+ gearman==2.0.2
  - \+ lxml==3.5.0
  - \+ requests==2.7.0
  - \+ unidecode==0.04.19
#### MCPServer:
- debian/control
  - \- python-pyinotify **(unused)**
  - \- python-gearman **(moved to requirements.txt)**
  - \- python-mysqldb **(moved to requirements.txt)**
  - \- python-lxml **(moved to requirements.txt)**
- requirements.txt
  - \+ Django>=1.7,<1.8
  - \+ django-mysqlpool==0.1-9
  - \+ django-extensions==1.1.1
  - \+ MySQL-python==1.2.5
  - \+ gearman==2.0.2
  - \+ lxml==3.5.0
#### archivematicaCommon:
- debian/control
  - \- python2.7-elementtree **(unneeded)**
  - \- python-mimeparse **(unused)**
  - \- python-dateutil **(moved to requirements.txt)**
- requirements.txt
  - \+ python-dateutil==2.4.2

### CODE REVIEW

File: src/MCPClient/requirements/base.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Django>=1.7,<1.8
+django-mysqlpool==0.1-9
+django-extensions==1.1.1
+MySQL-python==1.2.5
+gearman==2.0.2
+lxml==3.5.0
```
Comment (Hwesta) [inline src/MCPClient/requirements/base.txt:9]: We also need bagit==1.5.2, requests==2.7.0

File: src/MCPServer/debian/control [hunk 2]
```diff
@@ -15,7 +15,6 @@ Depends:
 	archivematica-dashboard(>=1.2.0),
 	dbconfig-common,
 	mysql-server,
-	python-pyinotify,
 	python-gearman,
 	python-mysqldb,
 	python-lxml,
```
Comment (Hwesta) [inline src/MCPServer/debian/control]: If we're installing lxml, mysql-python & gearman above, do we need the python-gearman, python-mysqldb, python-lxml still?

File: src/MCPClient/requirements/base.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Django>=1.7,<1.8
+django-mysqlpool==0.1-9
+django-extensions==1.1.1
+MySQL-python==1.2.5
```
Comment (Hwesta) [inline src/MCPClient/requirements/base.txt:7]: Can we switch to pymysql instead of this? It supports Python3, and is backwards compatible.  We already use it in agentarchives.

File: src/MCPServer/debian/control (hunk 2 above)
Comment (sevein) [inline src/MCPServer/debian/control]: No, you're right. I'll update the debian files accordingly.

File: src/archivematicaCommon/debian/control [hunk 4]
```diff
@@ -8,6 +8,6 @@ Homepage: http://archivematica.org
 
 Package: archivematica-common
 Architecture: i386 amd64 
-Depends: ${shlibs:Depends}, ${misc:Depends}, python, python-pip, python2.7-elementtree, python-mimeparse, python-dateutil
+Depends: ${shlibs:Depends}, ${misc:Depends}, python, python-pip, python2.7-elementtree, python-dateutil
```
Comment (Hwesta) [inline src/archivematicaCommon/debian/control]: Do we still need `python2.7-elementtree, python-dateutil`? Are they still used and/or can they be installed from pip?

File: src/dashboard/debian/control [hunk 5]
```diff
@@ -16,7 +16,6 @@ Depends:
 	python-pip,
 	python-dateutil,
 	python-gearman,
-	python-simplejson
```
Comment (mistydemeo) [inline src/dashboard/debian/control:19]: Oh wow, I didn't realize that was still in the Debian control file.

File: src/MCPClient/requirements/base.txt (hunk 3 above)
Comment (sevein) [inline src/MCPClient/requirements/base.txt:7]: I promise I'll try that but in a different PR please!

Comment (Hwesta) [inline src/MCPClient/requirements/base.txt:7]: Makes sense - I thought it was a drop-in replacement but it looks like it's not.

File: src/MCPClient/requirements/base.txt (hunk 1 above)
Comment (sevein) [inline src/MCPClient/requirements/base.txt:9]: Fixed.

File: src/MCPClient/requirements/base.txt (hunk 3 above)
Comment (sevein) [inline src/MCPClient/requirements/base.txt:7]: I submitted another pull request to discuss further, see #400.

**Comment by Hwesta** [comment]
:+1: but I'd like a second opinion to confirm it works and installs all the right things to the right places @mistydemeo @jhsimpson ?

**Comment by jhsimpson** [comment]
:+1:   

I looked over this and it looks good to me.  The only issue I spotted was about simplejson, which is removed from the debian control file in one place and not added back to requirements.txt, but it does not appear that is an actual problem.

**Comment by sevein** [comment]
Cool, thanks! b6eecbfff776ad61b5711230e788f9c6ef2b1729

**Comment by Hwesta** [comment]
> The only issue I spotted was about simplejson, which is removed from the debian control file in one place and not added back to requirements.txt, but it does not appear that is an actual problem. 

Simplejson is deprecated and not used anywhere, so removing it is fine. Json is built into stdlib now.

**Comment by sevein** [comment]
BTW we don't really have a mechanism to uninstall packages previously installed with apt. I'm guessing that's not a deal breaker, but I've seen cases in the past where a package made available by both apt and pip would conflict (py.test?). It may be a good idea to rebuild your development environment at some point.

**Comment by sevein** [comment]
Or run:

```
$ sudo apt-get purge python-dateutil python-gearman python-simplejson python-lxml python-mysqldb python-unidecode python2.7-elementtree python-mimeparse
```
